src/device.hpp

```diff
@@ -217,13 +217,32 @@ struct cvk_device : public _cl_device_id,
         return size * percentage_of_available_memory_reported;
     }
 
-    uint64_t memory_size() const {
-        return std::min(max_alloc_size() * 4, actual_memory_size());
-    }
+    uint64_t max_mem_alloc_size() const {
+        // Min memory as per the specs.
+        auto specMinAllocSz =
+            std::max(std::min((uint64_t)(1024 * 1024 * 1024),
+                              (uint64_t)((1 / 4) * global_mem_size())),
```

Comment:
This code will perform the integer division `1 / 4` which will always be 0. You probably want to write `global_mem_size() / 4` instead and maybe add an assert checking that `global_mem_size()` returns a multiple of ':

```
CVK_ASSERT(global_mem_size() % 4 == 0);
```